tests/integration/test_cogs_rates.py

```diff
@@ -101,6 +139,21 @@ async def test_compare_and_notify_all(rates_cog):
     assert dpytest.verify().message().nothing()
 
 
+@pytest.mark.asyncio
+async def test_compare_and_notify_all_exceptions(rates_cog, caplog):
+    # bad url, can't fetch
+    rates_cog.webpage_urls = ["http://localhost:5000/bogus-url.txt"]
+    await rates_cog.compare_and_notify_all()
+    with caplog.at_level(logging.ERROR):
+        assert "Could not retrieve rates from ARK Web API at" in caplog.text
+
+    # other issue, can't process
+    rates_cog.webpage_urls = ["asdfasdfasdfasdfasdf"]
+    await rates_cog.compare_and_notify_all()
+    with caplog.at_level(logging.ERROR):
+        assert "Could not retrieve rates from ARK Web API at" in caplog.text
```

Comment:
So I'm actually quite torn on this one. I was able to produce the exception here, and therefore we have test coverage for that exception handling block that raises it.

But I honestly kinda think it shouldn't have raised this exception. I almost want to fix the `get_current_rates()` method (which is ultimately the source of these exceptions), so that it raises something closer to what we'd expect.

Or at least better define what those exceptions are supposed to represent.

Reply:
This presentation provides no reason not to take that approach.

Reply:
#58